### ISSUE TITLE
Warning at #[derive(Deserialize)] when compiling without prelude

### ISSUE THREAD

**Issue by Cyres** [serde-rs/serde#1103]
Hi,

When using #[derive(Deserialize)] when compiling with #![no_implicit_prelude], there is a warning about "None". This will be hidden, if `use std::option::Option::None;` is used, but this shouldn't be required.

```
warning: variable `None` should have a snake case name such as `none`
   --> [...].rs:212:88
    |
212 | #[derive(Serialize, Deserialize)]
    |                     ^^^^^^^^^^^
```

Update: there is also an error with `Option::Some`.

**Comment by dtolnay** [comment]
Thanks! I released 1.0.22 with a fix for at least one bug that would have triggered this. If it still doesn't work with 1.0.22, could you share the full definition of the struct on line 212?

**Comment by Cyres** [comment]
It's now working, thank you!
